physionet-django/events/views.py

```diff
@@ -310,3 +314,60 @@ def event_detail(request, event_slug):
          'is_waitlisted': is_waitlisted,
          'event_datasets': event_datasets,
          })
+
+
+@login_required
+def manage_co_hosts(request, event_slug):
```

Comment:
There's a little bit too much logic in this view. I don't think the codebase has a notion of a "service" at this point, but ideally the view should just be parsing the request, dispatching some logic via a service and then preparing the response.

Here, it's parsing the request, handling errors, sending notifications, removing cohosts, adding cohosts etc. It's very hard to reason about this code. Maybe we should think about a separate module where this would be placed.

Reply:
As Discussed during the meeting - this needs to be addressed in a new Issue more generic on refactoring of codebase to reduce fat views and have service modules. I will document the discussion briefly and open up a new issue.
Resolving the comment here @kshalot